pennylane/transforms/defer_measurements.py

```diff
@@ -266,9 +268,20 @@ def func(x, y):
 
     for mp in tape.measurements:
         if mp.mv is not None:
-            # Update measurement value wires
-            wire_map = {m.wires[0]: control_wires[m.id] for m in mp.mv.measurements}
-            mp = qml.map_wires(mp, wire_map=wire_map)
+            # Update measurement value wires. We can't use `qml.map_wires` because the same
+            # wire can map to different control wires when multiple mid-circuit measurements
+            # are made on the same wire. This mapping is determined by the id of the
+            # MidMeasureMPs. Thus, we need to manually map wires for each MidMeasureMP.
+            if isinstance(mp.mv, MeasurementValue):
+                mp.mv.measurements = [
+                    qml.map_wires(m, {m.wires[0]: control_wires[m.id]}) for m in mp.mv.measurements
+                ]
+            else:
+                for val in mp.mv:
+                    val.measurements = [
+                        qml.map_wires(m, {m.wires[0]: control_wires[m.id]})
+                        for m in val.measurements
+                    ]
```

Comment:
Is there a non-in-place-modification way we could do this?

```
new_ms = [qml.map_wires(m, {m.wires[0]: control_wires[m.id]}) for m in mp.mv.measurements ]
new_m = MeasurementValue(new_ms, mp.mv.processing_fn)
mp = type(mp)(obs=new_m)
```
or something?  Mabye just add in a `copy` of `mp`?